### ISSUE TITLE
TFLite "Unrecognized GetAddress selector"

### ISSUE THREAD

**Issue by Kepnu4** [tensorflow/tensorflow#58402]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

2.10.0, master

### Custom Code

No

### Mobile device

iOS

### Current Behaviour?

```shell
The simple convolution net is not working with batch size > 1 on metal backend.
It works fine if convert with batch size = 1 and call multiply inferences or if inference on CPU
The problem exists on 2.10.0 and current master

It fails with log

VERBOSE: Replacing 62 node(s) with delegate (TfLiteMetalDelegate) node, yielding 1 partitions.
Unrecognized GetAddress selector
```
```


### Standalone code to reproduce the issue

```shell
-
```

### Relevant log output

_No response_</details>

**Comment by mohantym** [comment]
Hi @Kepnu4 !

Could you confirm that you have followed [CoreML delegate documentation](https://www.tensorflow.org/lite/ios/delegates/gpu#quantized_models)  and used a static batch size instead of 
 dynamic batch size for IOS.

[Ref ](https://github.com/tensorflow/tensorflow/issues/56835#issuecomment-1192223899)
A Quantized lite model is always recommended for GPU delegate usage. 

Would it be possible to share a reproducible code, device details along a sample lite model for replicating this issue.
 
Thank you!

**Comment by Kepnu4** [comment]
Hi @mohantym 
I use metal delegate with static input size (i.e batch size, width, height, channels).

Here is the model 
[model.tflite.zip](https://github.com/tensorflow/tensorflow/files/9916938/model.tflite.zip)

**Comment by bhack** [comment]
@Kepnu4 What static input size have you used?

https://github.com/tensorflow/tensorflow/issues/37012#issuecomment-652682677

**Comment by Kepnu4** [comment]
> @Kepnu4 What static input size have you used?
> 
> [#37012 (comment)](https://github.com/tensorflow/tensorflow/issues/37012#issuecomment-652682677)

It's `[3x256x256x5]`. I have static batch size  because channel is a sample in my case and it's always 3. If i convert with static size `[1x256x256x5]` and pass channel by channel then it works fine

**Comment by bhack** [comment]
Isn't expected that the layout is `B, C, H, W`?

**Comment by bhack** [comment]
Check https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/g3doc/performance/coreml_delegate.md#supported-ops

**Comment by Kepnu4** [comment]
> Check the layouts at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/g3doc/performance/coreml_delegate.md#supported-ops

I don't use CoreML delegate. I use metal delegate

**Comment by Kepnu4** [comment]
> Looking at the Metal Convolution tests the layout is the same: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/metal/kernels/conv_test.mm

Where do you see it? As see `BHWC` layout:
```
    op_def.src_tensors.push_back({data_type, TensorStorageType::BUFFER, Layout::BHWC});
    op_def.dst_tensors.push_back({data_type, TensorStorageType::BUFFER, Layout::BHWC});
    ConvGeneric operation = CreateConvGeneric(env->GetGpuInfo(), op_def, attr, &dst_shape);
    RETURN_IF_ERROR(env->ExecuteGPUOperation(src_tensor,
                                             std::make_unique<ConvGeneric>(std::move(operation)),
                                             dst_shape, &dst_tensor_ref));
```
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/metal/kernels/conv_test.mm#L262

**Comment by bhack** [comment]
> Where do you see it? As see BHWC layout:

No, I meant that looking at metal test is the same "as your"  `[3x256x256x5]`  `[1x256x256x5]`.

So `[1x256x256x5]` is working correctly and `[3x256x256x5]` is failing right?

**Comment by Kepnu4** [comment]
> @Kepnu4 !
> 
> Thanks for sharing your results. Could you check against [5x256x256x3] and let us know the results.
> 
> Thank you!

i can't shuffle channels because i will need to change network architecture 



> > Where do you see it? As see BHWC layout:
> 
> No, I meant that looking at metal test is the same "as your" `[3x256x256x5]` `[1x256x256x5]`.
> 
> So `[1x256x256x5]` is working correctly and `[3x256x256x5]` is failing right?

yes, right.

i can see that it's failing at his line with node name `"winograd_4x4_to_36 15"`
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/common/task/tensor_desc.cc#L981

**Comment by roserg** [comment]
Your model should work after this commit https://github.com/tensorflow/tensorflow/commit/89b2c189500f9537ddd958d247cead8fc555c575

**Comment by Kepnu4** [comment]
Thanks @roserg. It doesn't show error but gives wrong result with batch size > 1.
I tried to add batch to test and it seems that `Winograd4x4To36Test` gives zeros for all except first batch. Algthough i'm not sure if it is correct behaviour or not
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/common/tasks/winograd_test_util.cc#L165

**Comment by roserg** [comment]
Should be fixed after this commit https://github.com/tensorflow/tensorflow/commit/78cbae184cb2cd9541fb21074c3420b930c682ad

**Comment by Kepnu4** [comment]
> Should be fixed after this commit [78cbae1](https://github.com/tensorflow/tensorflow/commit/78cbae184cb2cd9541fb21074c3420b930c682ad)

It works correctly with this commit. Thanks @roserg

**Comment by mohantym** [comment]
Hi @Kepnu4 !
Could you confirm that whether the issue is still valid or not in 2.11 version. 
Thank you!
